### DISCUSSION TITLE
sqlboiler doesn't generate files for pivot tables in many to many relation

### DISCUSSION THREAD

**Discussion by rpg999** [aarondl/sqlboiler#1075] [Q&A] [answered]
I'm using Sqlboiler of version 4.8.4 to generate models. The data schema is provided below.

```
CREATE TABLE owner (
    address CHAR(42) NOT NULL,
    PRIMARY KEY (address)
);

CREATE TABLE owner_to_contract (
    owner_address CHAR(42) NOT NULL,
    contract_address CHAR(42) NOT NULL,
    PRIMARY KEY (owner_address, contract_address),
    CONSTRAINT fk_owner FOREIGN KEY (owner_address) REFERENCES owner(address),
    CONSTRAINT fk_contract FOREIGN KEY (contract_address) REFERENCES contract(address)
);

CREATE TABLE contract (
    address CHAR(42) NOT NULL,
    PRIMARY KEY (address)
);
```
 
I'm using Postgres SQL version 14.1

### Further information. What did you do, what did you expect?
I was expecting that SQLBoiler would create a file `owner_to_contract.go` for the table. Also, I was expecting it to create a helping struct with tables columns. For example

```
var OwnerToContractTableColumns = struct {
	OwnerAddress        string
	ContractAddress    string
}{
	OwnerAddress:         "owner_to_contract.owner_address",
	ContractAddress:     "owner_to_contract.contract_address",
}
```

Could you point me out if I need to specify any parameter on code generation to get that? Or it's what intended to be?
Thank you.

**Comment by stephenafamo** [comment, marked as answer]
This is on purpose. The many-to-many relationship is completely transparent.  
Instead, the `owner` struct has direct methods to query the `contracts` and vice versa.
 
Adding helpers to access the pivot tables is something that *may* be considered if there are strong use cases for it.

Is there a reason why you need models for the pivot table?

**Reply by rpg999** [reply]
The only thing I need is probably the generated table field names. It might be useful for writing `JOIN` queries. For now, my join queries look a bit bunky and I have to use strings for the fields in the pivot tables.
```golang
var qs []qm.QueryMod
qs = append(qs, qm.InnerJoin(
     models.TableNames.ContractToCollection+" ON "+models.TableNames.ContractToCollection+".collection_uid = "+models.CollectionTableColumns.UID,
))
qs = append(qs, qm.InnerJoin(
    models.TableNames.Contract+" ON "+models.TableNames.ContractToCollection+".contract_address = "+models.ContractTableColumns.Address,
))
```
If there were the helping struct with the field names, the query could be rewritten to use them only
```golang
qs = append(qs, qm.InnerJoin(
    models.TableNames.ContractToCollection+" ON "+models.ContractToCollectionTableColums.CollectionUID+" = "+models.CollectionTableColumns.UID,
))
```

**Reply by stephenafamo** [reply]
That makes sense. But this isn't very critical at the moment so it is unlikely to be added soon.
